backend/danswer/listeners/slack_listener.py

```diff
@@ -0,0 +1,193 @@
+    return "*Other potentially relevant documents:*\n" + top_documents_str
+
+
+def process_slack_event(client: SocketModeClient, req: SocketModeRequest) -> None:
+    if req.type == "events_api":
+        # Acknowledge the request anyway
+        response = SocketModeResponse(envelope_id=req.envelope_id)
+        client.send_socket_mode_response(response)
+
+        # Ensure that the message is a new message + of expected type
+        if req.payload.get("event", {}).get("type") != "message":
+            logger.info("Ignoring non-message event")
+
+        if req.payload.get("event", {}).get("subtype") is not None:
+            # this covers things like channel_join, channel_leave, etc.
+            logger.info("Ignoring message since is is a special message type")
+            return
+
+        if req.payload.get("event", {}).get("bot_profile"):
+            logger.info("Ignoring message from bot")
+            return
+
+        msg = req.payload.get("event", {}).get("text")
+        thread_ts = req.payload.get("event", {}).get("ts")
+        if not msg:
+            logger.error("Unable to process empty message")
+            return
+
+        # TODO: message should be enqueued and processed elsewhere,
+        # but doing it here for now for simplicity
+
+        def _get_answer(question: QuestionRequest) -> QAResponse | None:
+            try:
+                answer = answer_question(question=question, user=None)
+                if not answer.error_msg:
+                    return answer
+                else:
+                    raise RuntimeError(answer.error_msg)
+            except Exception as e:
+                logger.error(f"Unable to process message: {e}")
+            return None
+
+        answer = None
+        for _ in range(_NUM_RETRIES):
+            answer = _get_answer(
+                QuestionRequest(
+                    query=req.payload.get("event", {}).get("text"),
+                    collection=QDRANT_DEFAULT_COLLECTION,
+                    use_keyword=False,
+                    filters=None,
+                    offset=None,
+                )
+            )
+            if answer:
+                break
+
+        if not answer:
+            logger.error(
+                f"Unable to process message - did not successfully answer in {_NUM_RETRIES} attempts"
+            )
+            return
+
+        if not answer.answer:
+            logger.error(f"Unable to process message - no answer found")
+            return
+
+        # convert raw response into "nicely" formatted Slack message
+        quote_str, doc_identifiers = _process_quotes(answer.quotes)
+        top_documents_str = _process_documents(answer.top_ranked_docs, doc_identifiers)
+        if quote_str:
+            text = f"{answer.answer}\n\n*Sources:*\n{quote_str}\n\n{top_documents_str}"
+        else:
+            text = f"{answer.answer}\n\n*Warning*: no sources were quoted for this answer, so it may be unreliable 😔\n\n{top_documents_str}"
```

Comment:
Need to do some experimenting, I suspect with slack messages etc, gpt may correct typos etc, which is quite bad for us. Could you check if this happens? Just make a ticket and assign to me if it does :P I can take it up

Reply:
Not exactly sure what you mean by this 😅 Do you mean when giving back the quotes it will give it back with typos fixed? Why is this specific to slack messages?